src/lightning_app/runners/cloud.py

```diff
@@ -337,6 +340,20 @@ def dispatch(
                         default=True,
                     )
                     app_config.cluster_id = None
+                if existing_instance and existing_instance.spec.cluster_id != app_config.cluster_id:
+                    raise ValueError(
+                        dedent(
+                            f"""\
+                            Can not run app {app_config.name} on cluster {app_config.cluster_id} since it already exists on {existing_instance.spec.cluster_id}
+                                (moving apps between clusters is not supported).
+
+                            You can either:
+                                a. rename the app to run on {app_config.cluster_id} with the --name option
+                                    lightning run app script.py --name (new name) --cloud --cluster-id {app_config.cluster_id}
+                                b. delete the app running on {existing_instance.spec.cluster_id} in the UI before running this command.
```

Comment:
Let's add the exact line here `lightning delete app --name {app_config.name}`

Reply:
@lantiga the CLI doesn't have a `lightning delete app` command. We only have `lightning delete cluster`.

Reply:
We will soon: https://github.com/Lightning-AI/lightning/pull/15783
This should land tomorrow (correct @rlizzo ?) so let's add this one in view of having the command.